modules/ding_place2book/ding_place2book.fields.inc

```diff
@@ -566,9 +566,19 @@ function _ding_place2book_sync_p2b_entities($entity, array $settings) {
   $items = [];
 
   $event_id = !empty($settings['event_id']) ? $settings['event_id'] : FALSE;
+  $prices = $settings['prices_wrapper']['prices'];
   $event_maker_id = FALSE;
+
   // For clarity.
   $is_new = !$event_id;
+  if (arg(2) == 'clone') {
```

Comment:
Having a dependency on the page url deep down in a internal function does not seem like a good idea.

Is there something we can do using the clone api e.g. [`hook_clone_node_alter`](https://git.drupalcode.org/project/node_clone/blob/7.x-1.x/clone.api.php#L22)?

Reply:
The `hook_clone_node_alter` is called before the node form is rendered. As it is working now it is not possible to use this hook. As a solution we can store event id in a hidden form field and use it when make field presave(the POST/PUT/DELETE requests). This field will be populated on the node edit form. And on node clone this id will be removed so, the prices will be attached to the newly created entity.